cmd/snap-update-ns/bootstrap.c

```diff
@@ -368,6 +368,40 @@ void process_arguments(int argc, char *const *argv, const char **snap_name_out,
 				// Processing the user-fstab file implies we're being
 				// called from snap-confine.
 				should_setns = false;
+			} else if (!strcmp(arg, "-u")) {
+				const char *uid_text = argv[i + 1];
+				if (i + 1 == argc || uid_text == NULL) {
+					bootstrap_msg = "-u requires an argument";
+					bootstrap_errno = 0;
+					return;
+				}
+				errno = 0;
+				char *uid_text_end = NULL;
+				int parsed_uid = (int)strtol(uid_text, &uid_text_end, 10);
```

Comment:
`strtoul`?

Also, I think the string should be pre-validated first. `strto*` allow all kind of weird things, eg. `"    0x123", "0123", "\n123"` are valid strings for `strto*`. In contrast, we should allow only `[0-9]+`.

Reply:
I will expand the test coverage and see what happens. I think we validate enough because we ensure the whole string is parsed and we provide a base. If there's anything remaining it will indicate failure.

Reply:
I moved the code to `strtoul` and added way more tests now.